Mathlib/CategoryTheory/Functor/KanExtension/Adjunction.lean

```diff
@@ -136,6 +167,47 @@ noncomputable def lanCompColimIso (L : C ⥤ D) [∀ (G : C ⥤ H), L.HasLeftKan
       congr 1
       exact congr_app (L.lanUnit.naturality f) i))
 
+variable [HasColimitsOfShape D H]
+
+variable (G : C ⥤ H) [L.HasPointwiseLeftKanExtension G]
+
+omit [∀ (F : C ⥤ H), L.HasLeftKanExtension F]
+
+@[local instance]
+private theorem colimitIsoColimitGrothendieck_inst₁ :
+    HasColimit (CostructuredArrow.grothendieckProj L ⋙ G) :=
+  hasColimit_of_hasColimit_fiberwiseColimit_of_hasColimit _
+
+variable [HasColimitsOfShape C H]
+
+/-- If `G : C ⥤ H` is a left Kan extension of a functor `L : C ⥤ D` and `H` has colimits of shape
+`C`, `D`, and `CostructuredArrow L X` for all `X : D`, then the colimit of `G` is isomorphic to
+the colimit of a canonical functor `Grothendieck (CostructuredArrow.functor L) ⥤ H` induced by
+`L` and `G`. -/
+@[simps!]
+noncomputable def colimitIsoColimitGrothendieck :
```

Comment:
I wonder if this theorem really needs that the (non-pointwise) left Kan extension exists everywhere.

Reply:
Good point, I think we can avoid it actually, involves changing a few other definitions that unnecessarily assume it